src/com/massivecraft/vampire/cmd/CmdVampireDonate.java

```diff
@@ -0,0 +1,79 @@
+package com.massivecraft.vampire.cmd;
+
+import org.bukkit.inventory.PlayerInventory;
+
+import com.massivecraft.massivecore.MassiveException;
+import com.massivecraft.massivecore.command.requirement.RequirementHasPerm;
+import com.massivecraft.massivecore.command.requirement.RequirementIsPlayer;
+import com.massivecraft.massivecore.command.type.primitive.TypeDouble;
+import com.massivecraft.massivecore.util.MUtil;
+import com.massivecraft.vampire.BloodFlaskUtil;
+import com.massivecraft.vampire.Perm;
+import com.massivecraft.vampire.entity.MLang;
+
+public class CmdVampireDonate extends VCommand
+{
+	// -------------------------------------------- //
+	// CONSTRUCT
+	// -------------------------------------------- //
+	
+	public CmdVampireDonate()
+	{
+		// Aliases
+		this.addAliases("donate");
+		
+		// Parameters
+		this.addParameter(TypeDouble.get(), "amount", "4.0");
+		
+		// Requirements
+		this.addRequirements(RequirementHasPerm.get(Perm.DONATE));
+		this.addRequirements(RequirementIsPlayer.get());
+	}
+	
+	// -------------------------------------------- //
+	// OVERRIDE
+	// -------------------------------------------- //
+	
+	@Override
+	public void perform() throws MassiveException
+	{
+		double unlimitedAmount = this.readArg(4D);
+		
+		double amount = MUtil.limitNumber(unlimitedAmount, 0D, 20D);
+		if (amount != unlimitedAmount)
```

Comment:
This range checking should be within the type.

Please create a new Type extending TypeDouble and override the read method. Now you can cancel the command execution before coming to the spells and the  precondition logic is more separate :)
I can explain it to you more clearly in PM if you wish.